internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,311 @@
+		For(&kargoapi.ProjectConfig{}).
+		WithEventFilter(
+			predicate.Funcs{
+				DeleteFunc: func(event.DeleteEvent) bool {
+					return false
+				},
+			},
+		).
+		WithOptions(controller.CommonOptions(cfg.MaxConcurrentReconciles)).
+		Build(newReconciler(kargoMgr.GetClient(), cfg))
+	if err != nil {
+		return fmt.Errorf("error creating Project reconciler: %w", err)
+	}
+
+	logging.LoggerFromContext(ctx).Info(
+		"Initialized ProjectConfig reconciler",
+	)
+	return nil
+}
+
+func newReconciler(kubeClient client.Client, cfg ReconcilerConfig) *reconciler {
+	r := &reconciler{
+		cfg:    cfg,
+		client: kubeClient,
+	}
+	r.syncWebhookReceiversFn = r.syncWebhookReceivers
+	return r
+}
+
+// Reconcile is part of the main Kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *reconciler) Reconcile(
+	ctx context.Context,
+	req ctrl.Request,
+) (ctrl.Result, error) {
+	logger := logging.LoggerFromContext(ctx).WithValues(
+		"project-config", req.NamespacedName.Name,
+	)
+	ctx = logging.ContextWithLogger(ctx, logger)
+
+	// Fetch the ProjectConfig instance
+	projectConfig := &kargoapi.ProjectConfig{}
+	if err := r.client.Get(ctx, req.NamespacedName, projectConfig); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if projectConfig.DeletionTimestamp != nil {
+		logger.Debug("ProjectConfig is being deleted; nothing to do")
+		return ctrl.Result{}, nil
+	}
+
+	logger.Debug("reconciling ProjectConfig")
+	newStatus, needsRequeue, reconcileErr := r.syncProjectConfig(ctx, projectConfig)
+	logger.Debug("done reconciling ProjectConfig")
+
+	// Patch the status of the ProjectConfig.
+	if err := kubeclient.PatchStatus(ctx, r.client, projectConfig, func(status *kargoapi.ProjectConfigStatus) {
+		*status = newStatus
+	}); err != nil {
+		// Prioritize the reconcile error if it exists.
+		if reconcileErr != nil {
+			logger.Error(err, "failed to update ProjectConfig status after reconciliation error")
+			return ctrl.Result{}, reconcileErr
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to update ProjectConfig status: %w", err)
+	}
+
+	// Return the reconcile error if it exists.
+	if reconcileErr != nil {
+		return ctrl.Result{}, reconcileErr
+	}
+	// Immediate requeue if needed.
+	if needsRequeue {
+		return ctrl.Result{Requeue: true}, nil
+	}
+	// Otherwise, requeue after a delay.
+	// TODO: Make the requeue delay configurable.
+	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
+}
+
+func (r *reconciler) syncProjectConfig(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) (
+	kargoapi.ProjectConfigStatus,
+	bool,
+	error,
+) {
+	logger := logging.LoggerFromContext(ctx)
+	status := pc.Status.DeepCopy()
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReconciling,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
```

Comment:
A message is optional, and I wonder if it adds much here if the `Reason` has enough context.